### PR TITLE
Raise an error if there are no files available to run test

### PR DESCRIPTION
**Pull request by lexmag**
It helps in the following situation, for example:
`mix test run --no-start` or `mix test test/my_app_test.ex` (note `s` is missing)

I think we could list files given after the task name as well.

### CODE REVIEW

File: lib/mix/lib/mix/tasks/test.ex [hunk 1]
```diff
@@ -184,8 +184,12 @@ defmodule Mix.Tasks.Test do
     test_files   = parse_files(files, test_paths)
     test_pattern = project[:test_pattern] || "*_test.exs"
 
-    test_files = Mix.Utils.extract_files(test_files, test_pattern)
-    _ = Kernel.ParallelRequire.files(test_files)
+    case Mix.Utils.extract_files(test_files, test_pattern) do
+      [] ->
+        Mix.raise "No files available to compile"
```
Comment (josevalim) [inline lib/mix/lib/mix/tasks/test.ex]: "Test pattern did not match any file"?

Comment (josevalim) [inline lib/mix/lib/mix/tasks/test.ex]: If we can include the files (not the test_pattern) in the error message, even better!

Comment (lexmag) [inline lib/mix/lib/mix/tasks/test.ex]: How about this wording for a `mix test run missing.exs --no-start` command:
```
** (Mix) Test pattern did not match any file, the files given to compile: run, missing.exs
```
As I see none of the Mix errors are divided by newline.

Comment (josevalim) [inline lib/mix/lib/mix/tasks/test.ex]: sounds great.

**Comment by lexmag** [comment]
Updated.

**Comment by josevalim** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:
